clang/include/clang/Basic/AttributeCommonInfo.h

```diff
@@ -152,7 +152,12 @@ class AttributeCommonInfo {
     // FIXME: Eventually we want to do a list here populated via tablegen.  But
     // we want C++ attributes to be permissible on Lambdas, and get propagated
     // to the call operator declaration.
-    return getParsedKind() == AT_SYCLIntelKernelArgsRestrict;
+    auto ParsedAttr = getParsedKind();
+    if (ParsedAttr == AT_SYCLIntelKernelArgsRestrict ||
+        ParsedAttr == AT_SYCLIntelNumSimdWorkItems)
+      return true;
+
+    return false;
```

Comment:
I think this should be moved to the Tablegen level, as the `FIXME` above implies

Reply:
There is "but" part of the FIXME that makes me not doing it now.

Reply:
TBH, I don't understand this "but". How does marking attributes "lambda-specific" in Tablegen interfere with C++ attributes' propagation?

Reply:
Point here is that currently clang isn't accepting any right side function (function type) attributes, and attributes on lambda are part of this case (see: https://godbolt.org/z/z9khqk). For SYCL we added this functionality in clang. In the future, for a better good we will (I hope so) add this functionality to clang for C++ codegen in clang. But behavior of SYCL attributes and C++ attributes should be different. For SYCL we don't want an attribute to be propagated to a caller function. For C++ there is not such a restriction.

Reply:
Still your desire of applying the FIXME is understandable. But since future is vague and a FIXME resolution, that I can imagine is fragile, I don't really want to make this list generation in tablegen only to remove it in couple of months.

Reply:
Ah, sorry for not being clear from the beginning. Resolving this conversation, as the approach I thought about simply has no place in the surrounding code (below, if it's of interest).

I initially thought the following tablegen approach would sidestep the problem:
- add an intermediate class akin to `SYCLFunctionAttr` that would inherit `InheritableAttr`;
- supply it with a boolean field akin to `IsApplicableToLamdas` + a getter;
- inherit all SYCL function object attributes from `SYCLFunctionAttr` (that would otherwise be listed in this `if` condition). `IsApplicableToLambdas` could default to `1` in this intermediate base class.
Alas, this train of thought was bound to crash...